docs/api/types.md

```diff
@@ -0,0 +1,580 @@
+---
+sidebar_position: 4
+---
+
+# TypeScript Types
```

Comment:
not sure if we need a types reference

we are already exporting all the types in the library

so when people are coding with our library they would be able to see type definitions in their IDE

plus its just one more thing for us to maintain which might drift easily